signalfx/resource_signalfx_aws_external_integration.go

```diff
@@ -122,7 +124,68 @@ func integrationAWSExternalCreate(d *schema.ResourceData, meta interface{}) erro
 }
 
 func integrationAWSExternalDelete(d *schema.ResourceData, meta interface{}) error {
+	return DoIntegrationAWSDelete(d, meta)
+}
+
+// DoIntegrationAWSDelete is public because it is also used by integrationAWSTokenDelete
+func DoIntegrationAWSDelete(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*signalfxConfig)
 
+	// Retrieve current integration state
+	int, err := config.Client.GetAWSCloudWatchIntegration(context.TODO(), d.Id())
+	if err != nil {
+		return fmt.Errorf("Error fetching existing integration for integration %s, %s", d.Id(), err.Error())
+	}
+
+	// Only disable the Cloudwatch Metric Stream synchronization if needed
+	if int.MetricStreamsSyncState != "" && int.MetricStreamsSyncState != "DISABLED" {
+		int.MetricStreamsSyncState = "CANCELLING"
```

Comment:
Merging `signalfx_aws_external_integration` and `signalfx_aws_token_integration` resources into `signalfx_aws_integration` would avoid duplicating that code and simplify much things IMO. That would be a breaking change for existing configurations however.

I do not see any reason why they were separated in the first place.

Reply:
I agree, it looks like they could be merged in a backwards compatible way, though I'd rather do it in a separate PR. For now let's keep it as-is.